src/PrestaShopBundle/Service/Grid/ResponseBuilder.php

```diff
@@ -29,29 +29,47 @@
 use PrestaShop\PrestaShop\Core\Grid\Definition\Factory\GridDefinitionFactoryInterface;
 use PrestaShop\PrestaShop\Core\Grid\Definition\GridDefinitionInterface;
 use PrestaShop\PrestaShop\Core\Grid\Filter\GridFilterFormFactoryInterface;
+use PrestaShopBundle\Entity\Repository\AdminFilterRepository;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Routing\Router;
 
 class ResponseBuilder
 {
+    /** @var AdminFilterRepository */
+    private $adminFilterRepository;
+
+    /** @var int */
+    private $employeeId;
+
     /** @var GridFilterFormFactoryInterface */
     private $filterFormFactory;
 
     /** @var Router */
     private $router;
 
+    /** @var int */
+    private $shopId;
+
     /**
      * @param GridFilterFormFactoryInterface $filterFormFactory
      * @param Router $router
+     * @param AdminFilterRepository $adminFilterRepository
+     * @param int $shopId
      */
     public function __construct(
         GridFilterFormFactoryInterface $filterFormFactory,
-        Router $router
+        Router $router,
+        AdminFilterRepository $adminFilterRepository,
```

Comment:
Isn't it a BC break to modify constructor ? Can we avoid this will optional args :) ?

Reply:
Right. I document it.